File: CareConnect/CareConnect.Models/SearchObjects/EmployeeAvailabilitySearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class EmployeeAvailabilitySearchObject : BaseSearchObject<EmployeeAvailabilityAdditionalData>
    {
        public string? DayOfWeek { get; set; } = null!;

        public string? StartTime { get; set; }

        public string? EndTime { get; set; }

        public string? EmployeeFirstNameGTE { get; set; } = null!;

        public string? EmployeeLastNameGTE { get; set; } = null!;

        public string? ServiceNameGTE { get; set; } = null!;

        public int? EmployeeId { get; set; }

        public int? ServiceId { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/employee_availability_search_object.dart
import 'package:careconnect_mobile/models/search_objects/employee_availability_additional_data.dart';

import 'package:json_annotation/json_annotation.dart';

part 'employee_availability_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class EmployeeAvailabilitySearchObject {
  String? fts;
  String? dayOfWeek;
  String? startTime;
  String? endTime;
  String? employeeFirstNameGTE;
  String? employeeLastNameGTE;
  String? serviceNameGTE;
  int? employeeId;
  int? serviceId;
  int? page;
  String? sortBy;
  bool? sortAscending;

  bool? includeTotalCount;
  EmployeeAvailabilityAdditionalData? additionalData;

  EmployeeAvailabilitySearchObject({
    this.fts,
    this.dayOfWeek,
    this.startTime,
    this.endTime,
    this.employeeFirstNameGTE,
    this.employeeLastNameGTE,
    this.serviceNameGTE,
    this.employeeId,
    this.serviceId,
    this.page,
    this.sortBy,
    this.sortAscending,
    this.additionalData,
    this.includeTotalCount,
  });

  factory EmployeeAvailabilitySearchObject.fromJson(
    Map<String, dynamic> json,
  ) => _$EmployeeAvailabilitySearchObjectFromJson(json);

  Map<String, dynamic> toJson() =>
      _$EmployeeAvailabilitySearchObjectToJson(this);
}
